Mage.Sets/src/mage/cards/h/HinataDawnCrowned.java

```diff
@@ -0,0 +1,129 @@
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // Spells you cast cost {1} less to cast for each target.
+        this.addAbility(new SimpleStaticAbility(new HinataDawnCrownedOwnEffect()));
+
+        // Spells your opponents cast cost {1} more to cast for each target
+        this.addAbility(new SimpleStaticAbility(new HinataDawnCrownedOpponentsEffect()));
+    }
+
+    private HinataDawnCrowned(final HinataDawnCrowned card) {
+        super(card);
+    }
+
+    @Override
+    public HinataDawnCrowned copy() {
+        return new HinataDawnCrowned(this);
+    }
+}
+
+class HinataDawnCrownedOwnEffect extends CostModificationEffectImpl {
+
+    HinataDawnCrownedOwnEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral, CostModificationType.REDUCE_COST);
+        staticText = "Spells you cast cost {1} less to cast for each target";
+    }
+
+    private HinataDawnCrownedOwnEffect(HinataDawnCrownedOwnEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public HinataDawnCrownedOwnEffect copy() {
+        return new HinataDawnCrownedOwnEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        Set<UUID> targets = null;
+        if (game.inCheckPlayableState()) {
+            targets = CardUtil.getAllPossibleTargets(abilityToModify, game);
+        } else {
+            targets = CardUtil.getAllSelectedTargets(abilityToModify, game);
+        }
+        for (int i = 0; i < targets.stream().count(); i++) {
+            CardUtil.reduceCost(abilityToModify, 1);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean applies(Ability abilityToModify, Ability source, Game game) {
+        return abilityToModify instanceof SpellAbility
+                && abilityToModify.isControlledBy(source.getControllerId());
+    }
+}
+
+class HinataDawnCrownedOpponentsEffect extends CostModificationEffectImpl {
+
+    HinataDawnCrownedOpponentsEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral, CostModificationType.INCREASE_COST);
+        staticText = "Spells your opponents cast cost {1} more to cast for each target";
+    }
+
+    private HinataDawnCrownedOpponentsEffect(HinataDawnCrownedOpponentsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public HinataDawnCrownedOpponentsEffect copy() {
+        return new HinataDawnCrownedOpponentsEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        Set<UUID> targets = null;
+        if (game.inCheckPlayableState()) {
+            targets = CardUtil.getAllPossibleTargets(abilityToModify, game);
+        } else {
+            targets = CardUtil.getAllSelectedTargets(abilityToModify, game);
+        }
+        for (int i = 0; i < targets.stream().count(); i++) {
+            CardUtil.increaseCost(abilityToModify, 1);
```

Comment:
Same deal here:

CardUtil.increaseCost(abilityToModify, targets.stream().count());

 I will also make that change if that does in fact work